src/vl-upload.js

```diff
@@ -252,6 +253,20 @@ export class VlUpload extends vlElement(HTMLElement) {
     this._button.focus();
   }
 
+  /**
+   * Enable input element.
+   */
+  enable() {
+    vl.upload.enable(this._element);
+  }
+
+  /**
+   * Disable input element.
+   */
+  disable() {
+    vl.upload.disable(this._element);
+  }
```

Comment:
Ik heb ineens ook de test uitgevoerd in onze VHRM Controle Viewer applicatie. Als ik het attribuut zet resulteert dit in de upload die afgebroken wordt (en je ziet niet meer welk bestand er gekozen was). Dit is niet het gewenste gedrag (zie #61 voor meer info).

Zoals het nu is heeft ook wel nut voor bepaalde use cases, je kan verhinderen dat iemand een upload start. Maar wat wij wouden is interactie tijdens het uploaden onmogelijk maken.

Reply:
Klopt dat centraal voorzien werd om bij een disable de pending uploads te annuleren. Is het mogelijk om meer informatie te geven over de use-case (in groter geheel) a.d.h.v. mockups of screenshots? Zo kunnen we bepalen of er nood is aan een disabled upload die alsnog de pending uploads verwerkt of er eerder een ander UI/UX patroon nodig is.

Reply:
De use case is als volgt bij ons in de VHRM Controle Viewer:

We laden een bestand met controles op dat momenteel synchroon verwerkt wordt. Afhankelijk van de data doen we een call naar een externe service. Hierdoor kan het zijn dat bij veel data het verwerken iets langer duurt, vb 30 seconden.

Je wil niet dat de gebruiker vb direct tijdens opladen nog met die vl-upload zit te prutsen. Dus zou die inactief moeten zijn of iets tonen zoals vl-ui-loader of misschien soort van progress bar(s) van upload(s).

Momenteel is onze opladen knop wel inactief en toont die aan de het opladen bezig is, maar het lijkt mij beter dat de volledige vl-ui-upload een indicatie heeft. We zetten nu wel manueel alle knoppen (kruisje, ...) in de vl-ui-upload af zodat de gebruiker die niet meer kan manipuleren.

Hier een screenshot:

![image](https://user-images.githubusercontent.com/29454303/103355276-275b6c80-4aae-11eb-8c3f-34f111f461c1.png)

Voor onze use case zouden we naar een asynchrone verwerking kunnen gaan zodat de oplaadtijd minimaal wordt, maar ik denk dat er andere use cases zijn waarbij v.b. grote bestanden worden opgeladen waardoor het opladen tijd in beslag kan nemen.

Reply:
Kan je nog een screenshot bezorgen van het groter geheel. Het is nu bijvoorbeeld moeilijk om in te schatten of in deze use-case er nog input elementen aanwezig zijn.

Reply:
![image](https://user-images.githubusercontent.com/29454303/103356425-2d068180-4ab1-11eb-84d5-16bc484edb65.png)

Reply:
Na overleg met de huisstijl specialisten blijkt dat disabled hier eigenlijk niet voor geschikt is. Voorstel om op te vangen met de vl-ui-loader:

![upload](https://user-images.githubusercontent.com/11139249/104918555-38a0e600-5995-11eb-8e17-f528d8fd2349.png)

Reply:
OK, prima. Zal daar een ticket voor onze backlog voor maken. Had verwacht dat de upload zelf zo'n "loading" zou doen, maar als het "extern" is kunnen we inderdaad de "context" (dus ook de select, ...) vergroten.

Reply:
Ik stel voor om dit op niveau van vl-ui-form te implementeren, zodat een volledig formulier in een loading state kan komen.